metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,241 @@
+package scala.meta.internal.metals.codeactions
+
+import org.eclipse.lsp4j.{CodeActionParams, TextEdit}
+import org.eclipse.{lsp4j, lsp4j => l}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.meta.inputs.Position
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.{Buffers, CodeAction}
+import scala.meta.internal.parsing.Trees
+import scala.meta.internal.trees.Origin.Parsed
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+import scala.meta.{Defn, Pkg, Template, Term, Tree}
+
+class BracelessBracefulSwitchCodeAction(
+    trees: Trees,
+    buffers: Buffers
+) extends CodeAction {
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  def createCodeActionForBraceableTree(
+      hasBraces: (Tree, String) => Boolean,
+      path: AbsolutePath,
+      braceableTree: Tree,
+      braceableBranch: Tree,
+      bracelessStart: String,
+      bracelessEnd: String,
+      document: String
+  ) = {
+    if (hasBraces(braceableTree, document)) {
+      if (braceableTree.allowBracelessSyntax) {
+        // TODO `canUseBracelessSyntax` should be replaced by`canBeSyntacticallyBraceless`
+        // TODO Important: autoIndentDocument()
+        val braceableBranchLSPPos = braceableBranch.pos.toLSP
+        val braceableBranchStart = braceableBranchLSPPos.getStart
+        val braceableBranchEnd = braceableBranchLSPPos.getEnd
+        val startTextEdit = new TextEdit(
+          new l.Range(braceableBranchStart, braceableBranchStart),
+          bracelessStart
+        )
+        val endTextEdit = new TextEdit(
+          new l.Range(braceableBranchEnd, braceableBranchEnd),
+          bracelessEnd
+        )
+        val codeAction = new l.CodeAction()
+        codeAction.setTitle(BracelessBracefulSwitchCodeAction.goBraceless)
```

Comment:
We should probably make sure that the indentation is correct in this case,

Reply:
This PR absolutely needs the auto-indentation functionality for getting it right!